src/Nest/Search/Search/Highlighting/Highlight.cs

```diff
@@ -239,19 +230,17 @@ public class HighlightDescriptor<T> : DescriptorBase<HighlightDescriptor<T> ,IHi
 		public HighlightDescriptor<T> FragmentOffset(int fragmentOffset) => Assign(a => a.FragmentOffset = fragmentOffset);
 
 		// <inheritdoc/>
-		public HighlightDescriptor<T> Encoder(string encoder) => Assign(a => a.Encoder = encoder);
+		public HighlightDescriptor<T> Encoder(HighlighterEncoder? encoder) => Assign(a => a.Encoder = encoder);
 
 		// <inheritdoc/>
-		public HighlightDescriptor<T> Order(string order) => Assign(a => a.Order = order);
+		public HighlightDescriptor<T> Order(HighlighterOrder? order) => Assign(a => a.Order = order);
 
 		// <inheritdoc/>
-		public HighlightDescriptor<T> RequireFieldMatch(bool requireFieldMatch) => Assign(a => a.RequireFieldMatch = requireFieldMatch);
+		public HighlightDescriptor<T> RequireFieldMatch(bool requireFieldMatch = true) => Assign(a => a.RequireFieldMatch = requireFieldMatch);
```

Comment:
New default of true changes behaviour - is this intentional?

Reply:
Yes, it's intentional. Where methods take a `bool` parameter and the calling of the method is a positive intent (i.e. enables something), a default parameter value of `true` should be passed. I thought we had a test for this, but the test is for `bool?`:

https://github.com/elastic/elasticsearch-net/blob/5617675f61c368be30c93e710c9440adeec1289d/src/Tests/CodeStandards/Descriptors.doc.cs#L96

This should be updated to accept `bool?`. I think we also need a sweep before release to ensure we're adhering to this convention.